src/Text/LLVM/PP.hs

```diff
@@ -34,6 +34,46 @@ import Prelude hiding ((<>))
 
 -- Pretty-printer Config -------------------------------------------------------
 
+
+-- | The value used to specify the LLVM major version.  The LLVM text format
+-- (i.e. assembly code) changes with different versions of LLVM, so this value is
+-- used to select the version the output should be generated for.
+--
+-- At the current time, changes primarily occur when the LLVM major version
+-- changes, and this is expected to be the case going forward, so it is
+-- sufficient to reference the LLVM version by the single major version number.
+-- There is one exception and one possible future exception to this approach:
+--
+--  1. During LLVM v3, there were changes in 3.5, 3.6, 3.7, and 3.8.  There are
+--     explicit @ppLLVMnn@ function entry points for those versions, but in the
+--     event that a numerical value is needed, we note the serendipitous fact
+--     that prior to LLVM 4, there are exactly 4 versions we need to
+--     differentiate and can therefore assign the values of 0, 1, 2, and 3 to
+--     those versions (and we have no intention of supporting any other pre-4.0
+--     versions at this point).
+--
+--  2. If at some future date, there are text format changes associated with a
+--     minor version, then the LLVM version designation here will need to be
+--     enhanced and made more sophisticated.  At the present time, the likelihood
+--     of that is small enough that the current simple implementation is a
+--     benefit over a more complex mechanism that might not be needed.
+--
+type LLVMVer = Int
+
+-- | Helpers for specifying the LLVM versions prior to v4
+llvmV3_5, llvmV3_6, llvmV3_7, llvmV3_8 :: LLVMVer
+llvmV3_5 = 0
+llvmV3_6 = 1
+llvmV3_7 = 2
+llvmV3_8 = 3
+
+-- | This value should be updated when support is added for new LLVM versions;
+-- this is used for defaulting and otherwise reporting the maximum LLVM version
+-- known to be supported.
+llvmVlatest :: LLVMVer
+llvmVlatest = 16
```

Comment:
Is this used anywhere?

Reply:
It will be in forthcoming PR's (notably in llvm-pretty-bc-parser).